models.go

```diff
@@ -643,3 +643,19 @@ type ServerInfoRepesentation struct {
 	SystemInfo SystemInfoRepresentation `json:"systemInfo,omitempty"`
 	MemoryInfo MemoryInfoRepresentation `json:"memoryInfo"`
 }
+
+// IdentityProviderRepresentation represents an identity provider
+type IdentityProviderRepresentation struct {
+	AddReadTokenRoleOnCreate  bool              `json:"addReadTokenRoleOnCreate,omitempty"`
```

Comment:
Use pointers for all basic types such bool, string, int, etc...
`AddReadTokenRoleOnCreate  *bool` instead of `AddReadTokenRoleOnCreate  bool`

Take a look how it is done in the lates code